data/sqlutil/dynamic_frame.go

```diff
@@ -10,13 +10,25 @@ import (
 
 const STRING = "string"
 
-func isDynamic(converters []Converter) bool {
-	for _, conv := range converters {
+func isDynamic(converters []Converter) (bool, []Converter) {
+	dynamic := false
+	index := -1
+	for i, conv := range converters {
 		if conv.Dynamic {
-			return true
+			dynamic = true
+			index = i
+			break
 		}
 	}
-	return false
+	if dynamic {
+		// Remove the dynamic converter from the list, It's only used to determine if we should use the dynamic framer.
+		return true, remove(converters, index)
```

Comment:
this is removing the item from the input, which can be unexpected. Why not writing a simpler `filterDynamic` function?

```go
func filterDynamic(converters []Converter) []Converter {
	var dynamicConverters []Converter
	for _, conv := range converters {
		if conv.Dynamic {
			dynamicConverters = append(dynamicConverters, conv)
		}
	}
	return dynamicConverters
}
```